hubblestack_nova/firewall.py

```diff
@@ -118,9 +118,12 @@ def audit(data_list, tags, verbose=False):
         for tag_data in ret['Controlled']:
             tag = tag_data['tag']
             control_reason = tag_data.get('control', '')
-            if (tag, control_reason) not in control_reasons:
-                controlled.append({tag: control_reason})
-                control_reasons.add((tag, control_reason))
+            description = tag_data.get('description')
+            if (tag, description, control_reason) not in control_reasons:
+                dict = {'description': description,
```

Comment:
You shouldn't use the variable name `dict`, as that's a basic type. Overwriting it is a bad idea.

Use `tag_dict` instead, maybe? Multiple references in each of the files.